### PR TITLE
WIP | Initial project setup and lecture 1 - Scala 101

### PR DESCRIPTION
**Pull request by pheymann**
Hi all,
I created the basic project setup to generate slides and prepare exercises. I used [revealjs](https://github.com/hakimel/reveal.js/) together with [ScalaJS](https://www.scala-js.org/) to scrap some boilerplate. You can build your slide deck within Scala, compile it and get a running HTML side as result. We can expose that later on as GitHub page once all lectures are ready.

Take a look into `lecture1_scala_101` to get an idea how to create slides. If you need some helper function which is currently missing just add it to [SlideShowUtil](https://github.com/scalasummerschool/lectures/blob/init-and-lecture1-scala101/lectures-shared/src/main/scala/SlideshowUtil.scala).

If you want to take a look at the slides do the following:

```bash
sbt> project scala101-lecture
sbt> fastOptJS
```

```bash
open lecture1_scala_101/index.html
```

This should open a browser window with the slideshow.

### Scala 101
I put together an introduction into Scala language with the very basic syntax constructors we need to continue this course. Take a look at the slides to check if something is missing or not clear. 

I haven't proof-read it yet. So be ready to see some spelling/grammar errors 😆 .

Furthermore, I used [ScalaCheck](https://www.scalacheck.org/) to test the code students write during an exercise. The exercises are located at `exercise1_scala_101`.

### CODE REVIEW

**Comment by certainty** [comment]
This works for me mostly. What I'd change though is the logo in the center. I'd move it to somewhere else so that you have more space for the actual content.

**Comment by pheymann** [comment]
What do you mean by "logo in the center". The logo should only be in the center on the initial chapter page. But then it is dark and mixes with the background. On the other pages/slides, it should be in the top left corner.

File: lecture1_scala_101/src/main/scala/Lecture.scala [hunk 1]
```diff
@@ -0,0 +1,1425 @@
...
+      codeFragment("""
+        val gandalf = new Person("Gandalf", 2019)
+
+        gandalf.isOlder(new Person("Frodo", 33)) == isOlder(gandalf)(new Person("Frodo", 33))
+      """)
+    ),
+
+    slide(
+      "Classes: methods",
+      <.p("Methods are functions which have their surrounding object in scope and belong to the class. They also provide a means to get infix notation.")
```
Comment (certainty) [inline lecture1_scala_101/src/main/scala/Lecture.scala:900]: Maybe it's worth to very briefly talk about open recursion? Or connect back to this later when recursion is introduced.

Comment (certainty) [inline lecture1_scala_101/src/main/scala/Lecture.scala:900]: Maybe it's useful to think of methods being curried functions that are partially a applied to `this` and fields of the the object.

File: lecture1_scala_101/src/main/scala/Lecture.scala [hunk 2]
```diff
@@ -0,0 +1,1425 @@
...
+      """)
+    ),
+
+    slide(
+      "Conditional cases",
+      code("""
+        val person: Person = ???
+
+        person match {
+          // you can also use `&` and `|`
```
Comment (certainty) [inline lecture1_scala_101/src/main/scala/Lecture.scala:1261]: Maybe include a short example for this?

**Review by certainty — APPROVED** [review]
I left a few remarks regarding things that could be added but non of them are required IMHO.

LGTM!

File: lecture1_scala_101/src/main/scala/Lecture.scala (hunk 1 above)
Comment (pheymann) [reply lecture1_scala_101/src/main/scala/Lecture.scala:900]: already in the slides

**Pull request merged**